packages/remix-react/components.tsx

```diff
@@ -1248,16 +1248,29 @@ export interface SubmitFunction {
  *
  * @see https://remix.run/api/remix#usesubmit
  */
-export function useSubmit(): SubmitFunction {
-  return useSubmitImpl();
+export function useSubmit(options?: SubmitOptions): SubmitFunction {
+  let key = undefined; // no key for global submit
+  return useSubmitImpl(key, options);
 }
 
-let defaultMethod = "get";
-let defaultEncType = "application/x-www-form-urlencoded";
-
-export function useSubmitImpl(key?: string): SubmitFunction {
+const DEFAULT_METHOD = "get";
+const DEFAULT_ENC_TYPE = "application/x-www-form-urlencoded";
+
+// IDEA: remove the export here?
+export function useSubmitImpl(
```

Comment:
I'm not sure why this is exported, so I added a comment to remind me to ask.

Reply:
I think we can safely remove the export, as it was never intended to be a public API

Reply:
But it was public, so we would be breaking BC in a minor release. Let's keep it exported, and remove it in v2.

Reply:
We released things that were technically breaking in the past (mostly on `@remix-run/server-runtime`'s side) when they weren't meant to be public API, so I don't think we should care here either

Reply:
Maybe a good idea to add a TODO for that?

Reply:
Done 👍